### PR TITLE
[LLVM] Make compilable with LLVM-20

### PR DESCRIPTION
**Pull request by parsifal-47**
Hello All,
this PR adapts for LLVM-20 changes. Please feel free to let me know if there is a better way to do it.
Thank you!

### CODE REVIEW

**Comment by cbalint13** [comment]
Thank you @parsifal-47 !

This fix is welcome, it looks good to me.

**Review by cbalint13 — APPROVED** [review]
LGTM

File: src/target/llvm/codegen_llvm.cc [hunk 1]
```diff
@@ -169,7 +169,11 @@ void CodeGenLLVM::InitTarget() {
   llvm::TargetMachine* tm = llvm_target_->GetOrCreateTargetMachine();
   module_->setTargetTriple(tm->getTargetTriple().str());
   module_->setDataLayout(tm->createDataLayout());
+#if TVM_LLVM_VERSION >= 200
+  data_layout_ = std::make_unique<llvm::DataLayout>(module_->getDataLayout());
```
[the remark this answers is not in the document]
Comment (cbalint13) [reply src/target/llvm/codegen_llvm.cc:173]: * Please also address this warning:
```
src/target/llvm/codegen_llvm.cc:1019:32: warning: 'CreateGlobalStringPtr' is deprecated: Use CreateGlobalString instead [-Wdeprecated-declarations]
 1019 |   llvm::Value* str = builder_->CreateGlobalStringPtr(format);
      |                                ^~~~~~~~~~~~~~~~~~~~~
      |                                CreateGlobalString
/usr/lib64/llvm20/include/llvm/IR/IRBuilder.h:2018:3: note: 'CreateGlobalStringPtr' has been explicitly marked deprecated here
 2018 |   LLVM_DEPRECATED("Use CreateGlobalString instead", "CreateGlobalString")
      |   ^
```
Something like:
```
+#if TVM_LLVM_VERSION >= 200
+  llvm::Value* str = builder_->CreateGlobalString(format);
+#else
   llvm::Value* str = builder_->CreateGlobalStringPtr(format);
+#endif
```

**Review by cbalint13 — CHANGES_REQUESTED** [review]
Apologies for previous rushed approval, I tested the proposal on my local and found these.

File: src/target/llvm/codegen_llvm.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbalint13) [reply src/target/llvm/codegen_llvm.cc:173]: I am asking for a bit more conservative ways due to very frequent llvm changes, it helps keep the track a bit better across version deviations and maintainers. Thanks for understanding 🙏.

File: src/target/llvm/codegen_cpu.cc [hunk 2]
```diff
@@ -731,7 +731,11 @@ void CodeGenCPU::CreateStaticInit(const std::string& init_fname, const Stmt& bod
 llvm::Value* CodeGenCPU::GetPackedFuncHandle(const std::string& fname) {
   // We will store the packed function handle in global space.
   // Initialize it during the first call.
+#if TVM_LLVM_VERSION >= 200
+  llvm::DataLayout layout = module_->getDataLayout();
```
[the remark this answers is not in the document]
Comment (parsifal-47) [reply src/target/llvm/codegen_cpu.cc]: updated, thank you!

File: src/target/llvm/codegen_cpu.cc [hunk 3]
```diff
@@ -1297,7 +1301,11 @@ void CodeGenCPU::DefineFunctionRegistry(Array<String> func_names) {
   }
   llvm::ArrayType* t_tvm_crt_func_ptrs =
       llvm::ArrayType::get(ftype_tvm_backend_packed_c_func_->getPointerTo(), funcs.size());
+#if TVM_LLVM_VERSION >= 200
+  llvm::DataLayout layout = module_->getDataLayout();
```
[the remark this answers is not in the document]
Comment (parsifal-47) [reply src/target/llvm/codegen_cpu.cc]: updated, thank you!

File: src/target/llvm/codegen_llvm.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (parsifal-47) [reply src/target/llvm/codegen_llvm.cc]: no problem, I have resolved all deprecation warnings

**Comment by parsifal-47** [comment]
> Apologies for previous rushed approval, I tested the proposal on my local and found these.

no problem, I have resolved all deprecation warnings, but I am not very confident about `llvmGetPointerTo` macro, let me know what you think, thank you!

**Review by cbalint13 — APPROVED** [review]
LGTM, this really handles all the warnings for deprecation.

**Comment by cbalint13** [comment]
@parsifal-47 , thank you for the work !
Let's further make sure all the CI tasks are in green to be able to merge it.

**Pull request merged**